core/src/main/java/org/infinispan/security/mappers/ClusterRoleMapper.java

```diff
@@ -36,18 +37,20 @@ public class ClusterRoleMapper implements MutablePrincipalRoleMapper {
    private EmbeddedCacheManager cacheManager;
    private static final String CLUSTER_ROLE_MAPPER_CACHE = "org.infinispan.ROLES";
    private Cache<String, RoleSet> clusterRoleMap;
+   private Cache<String, RoleSet> clusterRoleReadMap;
 
    @Start
    void start() {
       clusterRoleMap = cacheManager.getCache(CLUSTER_ROLE_MAPPER_CACHE);
+      clusterRoleReadMap = clusterRoleMap.getAdvancedCache().withFlags(Flag.SKIP_CACHE_LOAD, Flag.CACHE_MODE_LOCAL);
    }
 
    @Override
    public Set<String> principalToRoles(Principal principal) {
```

Comment:
`org.infinispan.security.mappers.ClusterRoleMapper#list` needs some love too.
Maybe `principalToRoles()` should just do `return list(principal.getName());`?

Reply:
They have slightly different behaviours

Reply:
> They have slightly different behaviours. The first one is used by the `Authorizer`, the second one is used by the API to allow manipulation.

allows manipulation!? I hope not since it won't be replicated and will make the mapping inconsistent between nodes in the cluster.
Why can't they use the same method? They seem to return and do the exactly same thing. 

In case I'm wrong, `list()` could use `clusterRoleReadMap` too :)

Reply:
apologies, `list()` does not allow modification per se (that's done by `grant` and `deny`) but there is a subtle difference in the way an empty set is returned: we want to be explicit about it being empty in the API, whereas for the actual authorization we want to behave as if the mapping isn't present.